app/src/main/java/me/quadphase/qpdex/AdvancedSearch.java

```diff
@@ -0,0 +1,508 @@
+import android.view.MenuItem;
+import android.widget.Spinner;
+import android.content.Intent;
+
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import me.quadphase.qpdex.databaseAccess.PokemonFactory;
+import me.quadphase.qpdex.pokemon.Ability;
+import me.quadphase.qpdex.pokemon.EggGroup;
+import me.quadphase.qpdex.pokemon.MinimalPokemon;
+import me.quadphase.qpdex.pokemon.Type;
+
+
+/**
+ * Created by Thinesh on 31-Aug-15
+ *
+ * Class deals with the advanced search feature accessed from main menu
+ *
+ * Can filter pokemon by
+ *
+ * Type 1
+ * Type 2
+ * Ability
+ * Generation Available
+ * Egg Groups 1
+ * Egg Groups 2
+ * TODO: All 6 Base stats
+ * TODO: The ability to include/exclude Primal/Mega/Alternate Forms
+ *
+ */
+
+
+public class AdvancedSearch extends ActionBarActivity {
+
+
+    // pokemonFactory to access the various DB access functions
+    PokemonFactory pokemonFactory;
+
+    // Pokemon information to filter
+    MinimalPokemon[] allMinimalPokemon;
+
+    //Final Filtered List
+    MinimalPokemon [] filteredMinimalPokemon;
+    ArrayList<Integer> filteredNationalIds;
+    ArrayList<Integer> filteredUniqueIds;
+    //Contains either or
+    ArrayList<Integer> currentFilterPokemonArrayList;
+
+    // Arrays to fill Spinners
+    Type[] allTypes;
+    Ability[] allAbilities;
+    EggGroup[] allEggGroups;
+    String allGenerations [];
+
+    // User Selections
+    Type selectedType1;
+    Type selectedType2;
+    EggGroup selectedEggGroup1;
+    EggGroup selectedEggGroup2;
+    Ability selectedAbility;
+    int selectedGeneration;
+    int selectedHpLower;
+    int selectedHpGreater;
+    int selectedAttLower;
+    int selectedAttGreater;
+    int selectedDefGreater;
+    int selectedDefLower;
+    int selectedSpattLower;
+    int selectedSpattGreater;
+    int selectedSpdefLower;
+    int selectedSpdefGreater;
+    int selectedSpeedLower;
+    int selectedSpeedGreater;
+
+
+    // UI elements
+    Spinner type1Spinner;
+    Spinner type2Spinner;
+    Spinner abilitySpinner;
+    Spinner eggGroup1Spinner;
+    Spinner eggGroup2Spinner;
+    Spinner generationSpinner;
+
+    EditText hpLowerTextUI;
+    EditText hpGreaterTextUI;
+    EditText attLowerTextUI;
+    EditText attGreaterTextUI;
+    EditText defLowerTextUI;
+    EditText defGreaterTextUI;
+    EditText spattLowerTextUI;
+    EditText spattGreaterTextUI;
+    EditText spdefLowerTextUI;
+    EditText spdefGreaterTextUI;
+    EditText speedLowerTextUI;
+    EditText speedGreaterTextUI;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        long startTime = System.nanoTime();
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_advanced_search);
+
+        pokemonFactory = PokemonFactory.getPokemonFactory(this.getApplicationContext());

```

Comment:
A bit. It's calling this.getApplicationContext(). I don't know how heavy that method is, or how big the context is. Other than that, it's not going to hurt or confuse the PokemonFactory because it will just retrieve the singleton instance that already exists.